src/main/java/com/scaleunlimited/flinkcrawler/functions/UrlDBFunction.java

```diff
@@ -148,6 +149,20 @@ public Integer getValue() {
                     }
                 });
 
+        // Track the number of active URLs.
+        context.getMetricGroup().gauge(CrawlerMetrics.GAUGE_URLS_ACTIVE.toString(),
+                new Gauge<Integer>() {
+                    @Override
+                    public Integer getValue() {
+                        try {
+                            return _numActiveUrls.value();
```

Comment:
This is a state variable, so it only has the value for the current domain. Instead, have another regular variable that is incremented whenever _numActiveUrls is incremented. And add a TODO & an issue to sync up this value with state whenever we restore from state.